### PR TITLE
Add deepinfra groq openrouter perplexity

### PR DESCRIPTION
**Pull request by allainclair**
Add almost all models for **deepinfra groq openrouter perplexity**

perplexity missing `mixtral-8x22b-instruct`, auth 401 problem

### CODE REVIEW

File: adapters/provider_adapters/perplexity_sdk_chat_provider_adapter.py [hunk 1]
```diff
@@ -48,6 +58,12 @@ class PerplexityModel(Model):
         cost=Cost(prompt=0.2e-6, completion=0.2e-6),
         context_length=16384,
     ),
+    # PerplexityModel(  # Failing Auth
```
Comment (amydeng2000) [inline adapters/provider_adapters/perplexity_sdk_chat_provider_adapter.py:61]: let's delete the comments since it's not supported

**Review by amydeng2000 — CHANGES_REQUESTED** [review]
@amydeng2000 check output token limit for OpenRouter before approving this PR

File: adapters/provider_adapters/groq_sdk_chat_provider_adapter.py [hunk 2]
```diff
@@ -29,6 +29,18 @@ class GroqModel(Model):
         context_length=4096,
         vendor_name="meta-llama",
     ),
```
[the remark this answers is not in the document]
Comment (allainclair) [reply adapters/provider_adapters/groq_sdk_chat_provider_adapter.py]: Done

File: adapters/provider_adapters/perplexity_sdk_chat_provider_adapter.py (hunk 1 above)
Comment (allainclair) [reply adapters/provider_adapters/perplexity_sdk_chat_provider_adapter.py]: Done

File: adapters/provider_adapters/openrouter_sdk_chat_provider_adapter.py
Comment (amydeng2000) [inline adapters/provider_adapters/openrouter_sdk_chat_provider_adapter.py:1]: I tested all the supported models we integrated here and I think the completion_length limitation was old. None of these models seem to have a completion_length. Can you help to remove this entirely?

Comment (amydeng2000) [reply adapters/provider_adapters/openrouter_sdk_chat_provider_adapter.py:1]: After resolving this you can feel free to merge!

**Review by amydeng2000 — APPROVED** [review]

File: adapters/provider_adapters/openrouter_sdk_chat_provider_adapter.py
Comment (allainclair) [reply adapters/provider_adapters/openrouter_sdk_chat_provider_adapter.py:1]: Removed all `completion_length` of openrouter

**Pull request merged**